internal/nozzle/nozzle.go

```diff
@@ -198,24 +200,29 @@ func (n *Nozzle) postMetrics() {
 	n.mapLock.Unlock()
 
 	timestamp := time.Now().Unix()
+
 	for _, client := range n.ddClients {
 		// Add internal metrics
-		k, v := client.MakeInternalMetric("totalMessagesReceived", totalMessagesReceived, timestamp)
+		k, v := client.MakeInternalMetric("totalMessagesReceived", metric.GAUGE, totalMessagesReceived, timestamp)
+		metricsMap[k] = v
+		k, v = client.MakeInternalMetric("totalMetricsSent", metric.GAUGE, n.totalMetricsSent, timestamp)
+		metricsMap[k] = v
+		k, v = client.MakeInternalMetric("slowConsumerAlert", metric.GAUGE, atomic.LoadUint64(&n.slowConsumerAlert), timestamp)
 		metricsMap[k] = v
-		k, v = client.MakeInternalMetric("totalMetricsSent", n.totalMetricsSent, timestamp)
+		k, v = client.MakeInternalMetric("metrics.sent", metric.COUNT, n.metricsSent, timestamp)
 		metricsMap[k] = v
-		k, v = client.MakeInternalMetric("slowConsumerAlert", atomic.LoadUint64(&n.slowConsumerAlert), timestamp)
+		k, v = client.MakeInternalMetric("metrics.dropped", metric.COUNT, n.metricsDropped, timestamp)
 		metricsMap[k] = v
 
-		err := client.PostMetrics(metricsMap)
-		// NOTE: We don't need to have a retry logic since we don't return error on failure.
-		// However, current metrics may be lost.
-		if err != nil {
-			n.log.Errorf("Error posting metrics: %s\n\n", err)
-		}
+		unsentMetrics := client.PostMetrics(metricsMap)
+
+		n.metricsSent += uint64(len(metricsMap)) - unsentMetrics
+		n.metricsDropped += unsentMetrics
```

Comment:
You're never sending those values, since you reset right after. 
You need to send them separately